otsdaq/CodeEditor/CodeEditor.cc

```diff
@@ -380,6 +386,60 @@ void CodeEditor::getFileContent(cgicc::Cgicc& cgiIn, HttpXmlDocument* xmlOut)
 
 }  // end getFileContent()
 
+//==============================================================================
+// getFileGitURL
+void CodeEditor::getFileGitURL(cgicc::Cgicc& cgiIn, HttpXmlDocument* xmlOut)
+{
+	std::string path = CgiDataUtilities::getData(cgiIn, "path");
+	path             = safePathString(StringMacros::decodeURIComponent(path));
+	xmlOut->addTextElementToData("path", path);
+
+	std::string extension = CgiDataUtilities::getData(cgiIn, "ext");
+	if(extension == "ots")
+		extension = "";  // special handling of ots extension (to get bash script
+		                 // properly)
+	if(!( extension == "bin" || //allow bin for read-only
+		(path.length() > 4 && path.substr(path.length() - 4) == "/ots")))
+		extension = safeExtensionString(extension);
+	xmlOut->addTextElementToData("ext", extension);
+
+	std::string gitPath;
+	size_t      i;
+	if((i = path.find("$USER_DATA/")) == 0 || (i == 1 && path[0] == '/'))  // if leading / or without
+		gitPath = CodeEditor::getFileGitURL(
+		    CodeEditor::USER_DATA_PATH, path.substr(i + std::string("$USER_DATA/").size()) + (extension.size() ? "." : "") + extension);
+	else if((i = path.find("$OTSDAQ_DATA/")) == 0 || (i == 1 && path[0] == '/'))  // if leading / or without
+		gitPath = CodeEditor::getFileGitURL(
+		    CodeEditor::OTSDAQ_DATA_PATH, path.substr(std::string("/$OTSDAQ_DATA/").size()) + (extension.size() ? "." : "") + extension);
+	else
+		gitPath = CodeEditor::getFileGitURL(CodeEditor::SOURCE_BASE_PATH, path + (extension.size() ? "." : "") + extension);
+
+	xmlOut->addTextElementToData("gitPath", gitPath);
+
+}  // end getFileGitURL()
+
+//==============================================================================
+// getFileGitURL
+std::string CodeEditor::getFileGitURL(const std::string& basepath, const std::string& path)
+{
+
+		//TODO!
+		//TODO! request with linux exec to spack environment
```

Comment:
Can we have Name/Date with TODO comments so we can tell how stale they are? Even better would be a Github issue number/link